### PR TITLE
methods-modular_programs-easy-q7

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/methods/modular_programs/easy/q7/TwoMethods.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package methods.modular_programs.easy.q7;
+
+/*
+Write a Java Program that asks the user for two double variables and performs two instructions with them,
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/TwoMethods.java:4]: Change question text to:
`
Write a Java Program that asks the user for two double variables and performs two instructions with them, using two methods. Firstly, take the first variable and raise it to the second variable. Secondly, find the larger double variable and convert that double to radians (assuming it is in degrees). 
`

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    static Stream<Arguments> degreeCalcInputProvider() {
+        return Stream.of(
+                Arguments.of(2, 4, 0.06981317007977318),
+                Arguments.of(3, 3, 0.05235987755982989),
+                Arguments.of(8.3, 11.2, 0.1954768762233649),
+                Arguments.of(-7.2, 3, 0.05235987755982989),
+                Arguments.of(-3.6, 4, 0.06981317007977318),
+                Arguments.of(5.67, -9.6, 0.09896016858807849));
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:58]: Same thing here, there should be a 0 case

Comment (Hedgemon4) [reply src/methods/modular_programs/easy/q7/MainTest.java:58]: Also, you have the second number as the larger one most of the time, so just be careful about that

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    @MethodSource("raisedCalcInputProvider")
+    void correctRaisedCalcMethod(double base, double exponent, double raised) throws Throwable {
+        Object output = MethodUtil.invokeIfMethodExists(TwoMethods.class, "raisedCalc", new Object[]{base, exponent}, double.class, double.class);
+        CustomAssertions._assertEquals(raised, output, 0.001, "Your raisedCalc method does not correctly calculate the first variable raised to the second.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("degreeCalcInputProvider")
+    void correctDegreeCalcMethod(double first, double second, double radians) throws Throwable {
+        Object output = MethodUtil.invokeIfMethodExists(TwoMethods.class, "degreeCalc", new Object[]{first, second}, double.class, double.class);
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:81]: Change to MethodTest

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q7/TwoMethods.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package methods.modular_programs.easy.q7;
+
+/*
+Write a Java Program that asks the user for two double variables and performs two instructions with them, using two
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/TwoMethods.java:4]:
```suggestion
Write a Java Program that asks the user for two double variables and performs two calculations with them, using two
```

File: src/methods/modular_programs/easy/q7/TwoMethods.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+methods. Firstly, take the first variable and raise it to the second variable using a method called raisedCalc.
+Secondly, find the larger double variable and convert that double to radians (assuming it is in degrees) using a method
+called degreeCalc.
+
+Sample output:
+    Enter the first number:
+    5.23
+    Enter the second number:
+    17.95
+    The value of the first variable raised to the second variable is 7.890514371332084E12
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/TwoMethods.java:14]:
```suggestion
    The value of the first number raised to the second number is 7.890514371332084E12
```

File: src/methods/modular_programs/easy/q7/TwoMethods.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    The value of the greater variable converted from degrees to radians is 0.31328660073298215
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/TwoMethods.java:15]:
```suggestion
    The value of the larger number converted from degrees to radians is 0.31328660073298215
```

File: src/methods/modular_programs/easy/q7/TwoMethods.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Scanner;
+
+public class TwoMethods {
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter the first number: ");
+        double a = input.nextDouble();
+        System.out.println("Enter the second number: ");
+        double b = input.nextDouble();
+        System.out.println("The value of the first variable raised to the second variable is " + raisedCalc(a, b));
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/TwoMethods.java:27]: Change variable to number

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Parsons
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:20]: Add a space after the parsons label

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                Arguments.of(-7.2, 3, 0.05235987755982989),
+                Arguments.of(-3.6, 4, 0.06981317007977318),
+                Arguments.of(5.67, -9.6, 0.09896016858807849),
+                Arguments.of(25, 0, 0.4363323129985824),
+                Arguments.of(0, 13, 0.22689280275926285),
+                Arguments.of(0, 0, 0.0));
+    }
+
+    static Stream<Arguments> mainMethodInputProvider() {
+        return Stream.of(Arguments.of("4 5", 1024.0, 0.08726646259971647), Arguments.of("5.7 2.3", 54.76614451050491, 0.09948376736367678), Arguments.of("6.1 -4.7", 2.0368065149446233E-4, 0.1064650843716541));
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:68]: Put these onto separate lines like the other two

Comment (Hedgemon4) [reply src/methods/modular_programs/easy/q7/MainTest.java:68]: Also, pass two ints instead of one string

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    @ParameterizedTest
+    @MethodSource("raisedCalcInputProvider")
+    void correctRaisedCalcMethod(double base, double exponent, double raised) throws Throwable {
+        Object[][] arguments = {
+                {base, double.class},
+                {exponent, double.class}
+        };
+        MethodTest m = new MethodTest(TwoMethods.class, "raisedCalc", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(raised, output, 0.001, "Your raisedCalc method does not correctly calculate the first variable raised to the second.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:85]:
```suggestion
        CustomAssertions._assertEquals(raised, output, 0.00001, "Your raisedCalc method does not correctly calculate the first variable raised to the second.");
```

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+        MethodTest m = new MethodTest(TwoMethods.class, "degreeCalc", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(radians, output, 0.001, "Your degreeCalc method does not correctly convert the larger number to radians.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void printsOutputCorrectly(double first, double second, double raised, double radians) {
+        runWithInput(first + " " + second);
+        assertEquals(raised, Double.parseDouble(getItemByName("raised")), 0.001, "Your program does not properly display the raised value.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:104]:
```suggestion
        assertEquals(raised, Double.parseDouble(getItemByName("raised")), 0.00001, "Your program does not properly display the raised value.");
```

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        assertEquals(radians, Double.parseDouble(getItemByName("radians")), 0.001, "Your program does not properly display the converted radians.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:105]:
```suggestion
        assertEquals(radians, Double.parseDouble(getItemByName("radians")), 0.00001, "Your program does not properly display the converted radians.");
```

File: src/methods/modular_programs/easy/q7/TwoMethods.java [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Scanner;
+
+public class TwoMethods {
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
+        System.out.println("Enter the first number: ");
+        double a = input.nextDouble();
+        System.out.println("Enter the second number: ");
+        double b = input.nextDouble();
+        System.out.println("The value of the first number raised to the second variable is " + raisedCalc(a, b));
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/TwoMethods.java:27]:
```suggestion
        System.out.println("The value of the first number raised to the second number is " + raisedCalc(a, b));
```

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+                Arguments.of(0, 25, 0.0),
+                Arguments.of(0, 0, 1.0));
+    }
+
+    static Stream<Arguments> degreeCalcInputProvider() {
+        return Stream.of(
+                Arguments.of(4, 2, 0.06981317007977318),
+                Arguments.of(3, 3, 0.05235987755982989),
+                Arguments.of(8.3, 11.2, 0.1954768762233649),
+                Arguments.of(-7.2, 3, 0.05235987755982989),
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:60]: You should add a larger value so you get a larger value in radians as well

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+        MethodTest m = new MethodTest(TwoMethods.class, "degreeCalc", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(radians, output, 0.001, "Your degreeCalc method does not correctly convert the larger number to radians.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void printsOutputCorrectly(double first, double second, double raised, double radians) {
+        runWithInput(first + " " + second);
+        assertEquals(raised, Double.parseDouble(getItemByName("raised")), 0.00001, "Your program does not properly display the raised value.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:106]:
```suggestion
        assertEquals(raised, Double.parseDouble(getItemByName("raised")), 0.00001, "Your program does not correctly print the first number raised to the second number.");
```

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+        assertEquals(radians, Double.parseDouble(getItemByName("radians")), 0.00001, "Your program does not properly display the converted radians.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:107]:
```suggestion
        assertEquals(radians, Double.parseDouble(getItemByName("radians")), 0.00001, "Your program does not  correctly print the larger number converted to radians.");
```

File: src/methods/modular_programs/easy/q7/MainTest.java [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+    @ParameterizedTest
+    @MethodSource("raisedCalcInputProvider")
+    void correctRaisedCalcMethod(double base, double exponent, double raised) throws Throwable {
+        Object[][] arguments = {
+                {base, double.class},
+                {exponent, double.class}
+        };
+        MethodTest m = new MethodTest(TwoMethods.class, "raisedCalc", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(raised, output, 0.00001, "Your raisedCalc method does not correctly calculate the first variable raised to the second.");
```
Comment (Hedgemon4) [inline src/methods/modular_programs/easy/q7/MainTest.java:87]:
```suggestion
        CustomAssertions._assertEquals(raised, output, 0.00001, "Your raisedCalc method does not correctly calculate the first number raised to the second number.");
```

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]
